custom_components/bosch_alarm/device.py

```diff
@@ -2,11 +2,30 @@
 
 from .const import DOMAIN
 
-def device_info_from_panel(panel):
-    return DeviceInfo(
-        identifiers={(DOMAIN, panel.serial_number)},
-        name=f"Bosch {panel.model}",
-        manufacturer="Bosch Security Systems",
-        model=panel.model,
-        sw_version=panel.firmware_version,
-    )
+class PanelConnection:
+    def __init__(self, panel, unique_id, model):
+        self.unique_id = unique_id
+        self.model = model
+        self.panel = panel
+        self.on_connect = []
+        panel.connection_status_observer.attach(self.on_connection_status_change)
+
+    def on_connection_status_change(self):
+        if not self.panel.connection_status():
+            return
+        for on_connect_handler in self.on_connect:
```

Comment:
can 1-linify with `handler() for handler in self.on_connect`

Reply:
we would have to wrap it in a list, does that actually end up being cleaner?